scripts/nb-tester/qiskit_docs_notebook_tester/__init__.py

```diff
@@ -37,21 +38,64 @@
 matplotlib.set_loglevel("critical")
 """
 
-# If not submitting jobs, we also run this code before notebook execution to mock the real backend
-MOCKING_CODE = """\
+MOCKING_CODE = """
 import warnings
-from qiskit_ibm_runtime import QiskitRuntimeService
-from qiskit.providers.fake_provider import GenericBackendV2
-
-def patched_least_busy(self, *args, **kwarg):
-  return GenericBackendV2(num_qubits=6, control_flow=True)
-
-QiskitRuntimeService.least_busy = patched_least_busy
 
 warnings.filterwarnings("ignore", message="Options {.*} have no effect in local testing mode.")
 warnings.filterwarnings("ignore", message="Session is not supported in local testing mode or when using a simulator.")
+warnings.filterwarnings("ignore", message="Session is not supported in local testing mode or when using a simulator.")
 """
 
+def generate_backend_patch(
+    backend_name, 
+    provider: Literal["qiskit_ibm_runtime", "qiskit_fake_provider", "runtime_fake_provider"] = "qiskit_ibm_runtime", 
+    **kwargs
+):
+    """
+    generate code for fetching a custom backend to inject into a notebook
+    """
+
+    # Generates a set of arguments for QiskitRuntimeService using kwargs
+    qiskit_runtime_service_args = ", ".join([f"{arg}=\"{val}\"" if val else f"{arg}=None" for arg, val in kwargs.items()])
```

Comment:
No need for a list comprehension - you can use a generator expression. That's more efficient

```suggestion
    qiskit_runtime_service_args = ", ".join(f"{arg}=\"{val}\"" if val else f"{arg}=None" for arg, val in kwargs.items())
```

Reply:
Also this variable should only be defined in the block for qiskit_ibm_runtime because it's the only one that uses it